python_modules/dagster/dagster/core/execution/context/input.py

```diff
@@ -292,8 +292,9 @@ def asset_partitions_time_window(self) -> TimeWindow:
 
         partition_key_range = self.asset_partition_key_range
         return TimeWindow(
-            partitions_def.time_window_for_partition_key(partition_key_range.start).start,
-            partitions_def.time_window_for_partition_key(partition_key_range.end).end,
+            # mypy thinks partitions_def is <nothing> here because ????
+            partitions_def.time_window_for_partition_key(partition_key_range.start).start,  # type: ignore
```

Comment:
maybe cast instead? Could this cause weirdness further down the typing chain otherwise

Reply:
i think i tried that and it didn't work, will double check